internal/olm/operator/bundle/install.go

```diff
@@ -98,3 +185,137 @@ func (i *Install) setup(ctx context.Context) error {
 
 	return nil
 }
+
+// addBundleToIndexImage adds the bundle to an existing index image if the bundle is not already present in the index image.
+func addBundleToIndexImage(ctx context.Context, indexImage string, bundleDeclConfig *declarativeconfig.DeclarativeConfig) (*declarativeconfig.DeclarativeConfig, error) {
+	log.Infof("Rendering a File-Based Catalog of the Index Image %q", indexImage)
+	log.SetOutput(ioutil.Discard)
+	render := action.Render{
+		Refs: []string{indexImage},
+	}
+
+	imageDeclConfig, err := render.Run(ctx)
+	if err != nil {
+		log.Errorf("Error in rendering the index image %q: %v", indexImage, err)
+		return nil, err
+	}
+	log.SetOutput(os.Stdout)
+
+	if len(bundleDeclConfig.Bundles) < 0 {
+		return nil, fmt.Errorf("bundles rendered are less than 0 for the bundle image")
+	}
+
+	// check if the bundle exists in the index image.
+	if len(bundleDeclConfig.Channels) > 0 && len(bundleDeclConfig.Bundles) > 0 {
+		for _, channel := range imageDeclConfig.Channels {
+			// find the specific channel that the bundle needs to be inserted into.
+			if channel.Name == bundleDeclConfig.Channels[0].Name && channel.Package == bundleDeclConfig.Channels[0].Package {
+				// check if the CSV name is already present in the channel's entries.
+				for _, entry := range channel.Entries {
+					if entry.Name == bundleDeclConfig.Bundles[0].Name {
+						return nil, fmt.Errorf("bundle %q already present in the index image: %s", bundleDeclConfig.Bundles[0].Name, indexImage)
+					}
+				}
+
+				break // we only want to search through the specific channel.
+			}
+		}
+	}
+
+	if len(bundleDeclConfig.Bundles) > 0 && len(bundleDeclConfig.Channels) > 0 && len(bundleDeclConfig.Packages) > 0 {
+		imageDeclConfig.Bundles = append(imageDeclConfig.Bundles, bundleDeclConfig.Bundles[0])
+		imageDeclConfig.Channels = append(imageDeclConfig.Channels, bundleDeclConfig.Channels[0])
+		imageDeclConfig.Packages = append(imageDeclConfig.Packages, bundleDeclConfig.Packages[0])
```

Comment:
I'm trying to think of various possible cases for these two statements. 

For example, if we are attempting to insert a memcached bundle into a "stable" channel of an index image that already has the memcached bundle in an "alpha" channel, we wouldn't want to append the package blob since it will already be present. We'd only want to append the channel and bundle blobs. 

Do we need another check to see if a package already exists?

Reply:
@joelanford thoughts on the above scenario?

Reply:
If the bundle already exists in a different channel, then you theoretically already have the package and bundle blob in the FBC. All you have to do at that point is reference it in whatever channel(s) you want it to be in.

Reply:
I'm getting a little bit tripped up by the length checks for the various `bundleDeclConfig` Bundles, Channels, and Packages. Based on the code paths, it seems like a guarantee that those will each contain _exactly_ one package, channel, and bundle in that FBC.

If that's the case, is it possible to refactor things such that the `bundleDeclConfig` length checks can be removed?

Perhaps this function signature could accept individual `pkg declcfg.Package, ch declcfg.Channel, b declcfg.Bundle` parameters? Or a custom unexported struct that allows exactly one of each of those?

I think doing something along those lines would help make it more obvious what this function is doing.

Reply:
@joelanford, we initially had it without the checks, but wanted to eliminate _any_ possibility of an index out of bound exception for the array (i.e., `Bundles[0]`). But you're right, with the way the `bundleDeclConfig` object is constructed, this should not happen at all. @rashmigottipati, I'm okay with removing the length checks.

Reply:
@joelanford @VenkatRamaraju I like the idea of having a custom unexported struct that allows exactly one of each. I'll refactor it and get rid of the length checks.

Reply:
@joelanford @VenkatRamaraju Created an unexported struct and got rid of the length checks in [this commit](https://github.com/operator-framework/operator-sdk/pull/5671/commits/a73c3db7cf7bc92e279c6b1c1e446846e1e67944) based on suggestions.